### DISCUSSION TITLE
Add "Config Templates" during creating device in script.

### DISCUSSION THREAD

**Discussion by LuaLua10** [netbox-community/netbox#15416] [Help Wanted!] [answered]
Hi.
Is there any possibility to bind "Config Templates" during device creation in the script (code below)? I found it possible to bind the configuration template only to "Device role" and "Platform". There is such a possibility in the Web interface.

```
    device_config_template = MultiObjectVar(
        model=ConfigTemplate,
        label="Config Template",
        description="Name of Config Template.",
        required=False
    )
```

```
device = Device(
            name=data["device_name"],
            asset_tag=data["asset_tag"],
            role=data["role"],
            platform=data["platform"],
            device_type=data["device_type"],
            site=data["site"],
            location=data["location"],
            rack=data["rack"],
            status=data["status"],
            comments=data["comments"],
        )
        device.full_clean()
        device.save()
        device.tags.set(data["device_tags"])
        self.log_success(f"Created new switch: {device}")
```

**Comment by candlerb** [comment, marked as answer]
In the source code, class Device is [here](https://github.com/netbox-community/netbox/blob/v3.7.3/netbox/dcim/models/devices.py#L536-L543). It mixes in RenderConfigMixin from [here](https://github.com/netbox-community/netbox/blob/v3.7.3/netbox/dcim/models/mixins.py#L54-L75).

```
    config_template = models.ForeignKey(
        to='extras.ConfigTemplate',
        on_delete=models.PROTECT,
        related_name='%(class)ss',
        blank=True,
        null=True
    )
```

Therefore you should be able to set something like this:

```
device = Device(
    ...
    config_template = data["device_config_template"],
)
```

although it needs to be an ObjectVar not a MultiObjectVar (since it points to a single ConfigTemplate, not a set of them)

**Reply by LuaLua10** [reply]
Thanks, it's working.
